pravega-k8-log-collection/pravega-k8s-fluentbit-log.sh

```diff
@@ -0,0 +1,93 @@
+# Detect Pravega pods information in various K8 deployment (e.g. PKS / GKE)
+if [[ $i == *"po/"* ]] || [[ $i == *"pod/"* ]]; then
+
+    # Collect logs from "Running" pods, if pods are not in "Running" state it will collect describe information.
+    if [ "`echo $i | awk '{print $4}'`" == "Running" ]; then
+    {
+        fluentbit_pod=$(echo $i |  awk '-F'[/] '{print $2}'| awk '{print $1}')
+        namespace=$(echo $i | awk '{print $1}')
+        # Clean any pre-existing tar file in the fluentbit pod.
+        kubectl -n $namespace exec $fluentbit_pod -- bash -c "rm /tmp/$fluentbit_pod.tar.gz" || true
+        # Create a tar.gz file with all the logs contained in the fluentbit pod.
+        set +e
+        kubectl -n $namespace exec $fluentbit_pod -- bash -c "tar -czf /tmp/$fluentbit_pod.tar.gz /var/vcap/store/docker/docker/containers"
+        exitcode=$?
+        # Some logs may be being written during collection, which makes tar to exit with a non-zero code. Need to handle this case.
+        if [ "$exitcode" != "1" ] && [ "$exitcode" != "0" ]; then
+                exit $exitcode
+        fi
+        set -e
+        # Download the compressed file from the fluentbit pod.
+        kubectl cp $namespace/$fluentbit_pod:/tmp/$fluentbit_pod.tar.gz $output_dir
+        # Remove temporal compressed file from fluentbit pod.
+        kubectl -n $namespace exec $fluentbit_pod -- bash -c "rm /tmp/$fluentbit_pod.tar.gz"
+        # Untar the contents to provide a better structure of the resulting logs artifact.
+        tar -xzvf $output_dir/$fluentbit_pod.tar.gz -C $output_dir
+        mv $output_dir/var/vcap/store/docker/docker/containers/ $output_dir/$fluentbit_pod
+        # Delete useless files from downloaded fluentbit dirs (preserve only .log files and the directory structure).
+        for container_dir in $(find $output_dir/$fluentbit_pod -maxdepth 1 -mindepth 1 -type d)
+        do
+                echo $container_dir
+                mv $container_dir/*.log $container_dir/..
+                rm -rf $container_dir/*
+                mv $container_dir/../*.log $container_dir
+        done
+        # Relate the name of the pods being logged with the actual log data.
+        kubectl -n $namespace exec $fluentbit_pod -- ls /var/vcap/store/docker/docker/containers/ > $output_dir/container-lognames
+        kubectl -n $namespace exec $fluentbit_pod -- ls /var/log/containers/ > $output_dir/pod-lognames
+        for podlog in $(cat $output_dir/pod-lognames)
+        do
+                log_pod_name=$(echo $podlog | awk -F_ '{print $1}')
+                log_container_name=$(echo $podlog | awk -F- '{print $NF}' |  awk -F. '{print $1}')
+                # This will help to easily locate the logs by pod name.
+                mv $output_dir/$fluentbit_pod/$log_container_name  $output_dir/$fluentbit_pod/$log_pod_name || true
+
+        done
+        # Clean temporal files.
+        rm -rf $output_dir/$fluentbit_pod.tar.gz
+        rm -rf $output_dir/var
```

Comment:
It's nice to cleanup the temporary files like pod-lognames & container-lognames
```suggestion
        rm -rf $output_dir/var
        rm -rf $output_dir/pod-lognames
        rm -rf $output_dir/container-lognames
```